File: garbage/solution/garbage-sub_2.py
XY_LIMIT = 2000

# read input
N, W, H = map(int, input().split())
grid = [[0 for i in range(2*XY_LIMIT+1)] for j in range(2*XY_LIMIT+1)] # can be optimized
for i in range(N):
    x, y, w = map(int, input().split())
    grid[x+1][y+1] += w

# 2D prefix sum
for i in range(1, 2*XY_LIMIT+1):
    for j in range(1, 2*XY_LIMIT+1):
        grid[i][j] += grid[i-1][j] + grid[i][j-1] - grid[i-1][j-1]

# check all positions for lower left corner
result = 0
for i in range(XY_LIMIT):
    for j in range(XY_LIMIT):
        value = grid[i+W][j+H] - grid[i][j+H] - grid[i+W][j] + grid[i][j]
        result = max(result, value)
print(result)

File: garbage/submissions/partially_accepted/garbage-sub_2.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

#define XY_LIMIT 2000

int main() {
    // read input
    int N, W, H;
    cin >> N >> W >> H;
    ll grid[2*XY_LIMIT+1][2*XY_LIMIT+1];
    for (int i = 0; i < N; i++) {
        int x, y, w;
        cin >> x >> y >> w;
        grid[x+1][y+1] += w;
    }

    // 2D prefix sum
    for (int i = 1; i <= 2*XY_LIMIT; i++) {
        for (int j = 1; j <= 2*XY_LIMIT; j++) {
            grid[i][j] += grid[i-1][j] + grid[i][j-1] - grid[i-1][j-1];
        }
    }

    // check all positions for lower left corner
    ll result = 0;
    for (int i = 0; i < XY_LIMIT; i++) {
        for (int j = 0; j < XY_LIMIT; j++) {
            ll value = grid[i+W][j+H] - grid[i][j+H] - grid[i+W][j] + grid[i][j];
            result = max(result, value);
        }
    }
    cout << result << endl;
}
